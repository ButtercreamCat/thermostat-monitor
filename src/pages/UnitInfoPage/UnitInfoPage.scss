.unitInfo {
  width: 100vw;
  // height: 100vh;
  // overflow: hidden;
  display: flex;

  &_content {
    width: 70%;
    margin: 5%;
    box-sizing: content-box;
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
}

@media only screen and (max-width: 768px) {
  .unitInfo {
    width: 100%;
    height: auto;
    min-height: 100%;
    overflow: auto;
    position: relative;
    &_content {
      width: 100%;
      height: auto;
      min-height: calc(100vh - 180px);
      padding: 0;
      margin-top: 180px;
      justify-content: flex-start;
    }
  }
}
