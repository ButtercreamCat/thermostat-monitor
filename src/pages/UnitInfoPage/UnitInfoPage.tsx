import React, { Component } from "react";
import { RouteComponentProps, Switch, Route } from "react-router-dom";
import "./UnitInfoPage.scss";
import ThermostatPage from "../ThermostatPage/ThermostatPage";
import ChartPage from "../ChartPage/ChartPage";
import InfoSidebar from "../../components/InfoSidebar/InfoSidebar";

type TParams = { id: string };

class UnitInfoPage extends Component<RouteComponentProps<TParams>> {
  render() {
    const { match } = this.props;
    return (
      <div className="unitInfo">
        <InfoSidebar id={match.params.id} url={match.url}></InfoSidebar>

        <Switch>
          <Route path={`${match.path}/thermostat`}>
            <ThermostatPage></ThermostatPage>
          </Route>
          <Route path={`${match.path}/chart`}>
            <ChartPage></ChartPage>
          </Route>
        </Switch>
      </div>
    );
  }
}

export default UnitInfoPage;
