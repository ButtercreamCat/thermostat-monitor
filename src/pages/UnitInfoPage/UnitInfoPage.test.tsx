import React from "react";
import { shallow } from "enzyme";
import UnitInfoPage from "./UnitInfoPage";
import { createMemoryHistory, createLocation } from "history";
import { match as routerMatch } from "react-router";

const history = createMemoryHistory();
const path = "/unit/:id";

const match: routerMatch<{ id: string }> = {
  isExact: false,
  path,
  url: path.replace(":id", "110"),
  params: { id: "110" }
};

const location = createLocation(match.url);

test("should render UnitInfoPage correctly", () => {
  const wrapper = shallow(
    <UnitInfoPage
      history={history}
      location={location}
      match={match}
    ></UnitInfoPage>
  );
  expect(wrapper).toMatchSnapshot();
});
