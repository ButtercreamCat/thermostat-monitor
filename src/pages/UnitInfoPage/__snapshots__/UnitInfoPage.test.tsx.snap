// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UnitInfoPage correctly 1`] = `
<div
  className="unitInfo"
>
  <InfoSidebar
    id="110"
    url="/unit/110"
  />
  <Switch>
    <Route
      path="/unit/:id/thermostat"
    >
      <ThermostatPage />
    </Route>
    <Route
      path="/unit/:id/chart"
    >
      <ChartPage />
    </Route>
  </Switch>
</div>
`;
